nameable.rb

```diff
@@ -0,0 +1,37 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'Method not implemented'
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.upcase
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    if @nameable.correct_name.length > 10
+      trimd_string = ''
+      num = 0
+      while num < 10
+        trimd_string += @nameable.correct_name.chars('')[num]
+        num += 1
+      end
+      trimd_string
+    end
+    @nameable.correct_name
+  end
+end
```

Comment:
- [ ] I'm getting some errors with the syntax of this function, in line 30, the `chars` method doesn't expect any arguments, and you're passing an empty string: 

![image](https://github.com/addisalemseifu/library-app-ruby/assets/77069011/17ebb37e-5c11-4995-b327-06e388d861a2)

Please consider fixing this! 🤓 🔧 

- [ ] Also, the function is not returning what's expected:

![Screenshot from 2023-08-28 11-39-10](https://github.com/addisalemseifu/library-app-ruby/assets/77069011/fcba7688-d25e-4617-8325-8ac7f4f3b33a)

This is because inside your `if` statement you are not returning the `trimd_string`. So the return value of this function is always `@nameable.correct_name`. Please consider fixing this! 🤓 🔧 

Expected output: `Maximilian`.

```suggestion
class TrimmerDecorator < Decorator
  def correct_name
    if @nameable.correct_name.length > 10
      trimd_string = ''
      num = 0
      while num < 10
        trimd_string += @nameable.correct_name.chars[num]
        num += 1
      end
      return trimd_string
    end
    @nameable.correct_name
  end
end
```